src/person/person.textSearch.interface.ts

```diff
@@ -0,0 +1,15 @@
+import { IPerson } from './person.interface';
+
+export type PersonSearchQuery = Partial<{
+  status: string | string[];
+  entityType: string | string[];
+  'domainUsers.dataSource': string | string[];
+  rank: string | string[];
+  responsibility: string | string[];
+  hierarchyPath: string;
+  fullName: string;
+}>;
+
+export interface PersonTextSearch {
+  searchByQuery(query: PersonSearchQuery): Promise<IPerson[]>;
```

Comment:
maybe refactor to `searchByFullName(fullName, filters)` and make the `fullName` required